client-registry-api/src/main/java/uk/gov/di/authentication/clientregistry/services/ClientConfigValidationService.java

```diff
@@ -196,4 +217,8 @@ private boolean areClientLoCsValid(List<String> clientLoCs) {
         }
         return true;
     }
+
+    private boolean isValidIdTokenSigningAlgorithm(String idTokenSigningAlgorithm) {
+        return validIdTokenSigningAlgorithms.contains(idTokenSigningAlgorithm);
```

Comment:
You could probably retool this method to do all the Optional-based logic that's currently in the if-statements, and then the two callsites would become
something like
```java
if (isNotValidIdTokenSigningAlgorithm(updateRequest.getIdTokenSigningAlgorithm())) {
...
}
```

Reply:
Yeah, it's a bit of an odd pattern. Would help with readability. I'm not really sure why we don't throw an exception rather than returning an an error object. Both validation calls are already in a try catch.

Reply:
I'll open a small BAU PR to refactor this when I have a moment, but I think it's makes sense to leave it as is for now as it's consistent with what's already there.

Reply:
I've opened a separate RP for the above change: https://github.com/govuk-one-login/authentication-api/pull/4629. It'll require a rebase once this RP is merged.